lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -46,34 +47,101 @@ def compute(self, student, course_structure, scores_client, submissions_scores):
                 filter_func=possibly_scored,
                 start_node=self.location,
         ):
-            descendant = course_structure[descendant_key]
+            self._compute_block_score(student, descendant_key, course_structure, scores_client, submissions_scores)
 
-            if not getattr(descendant, 'has_score', False):
-                continue
+        self.all_total, self.graded_total = graders.aggregate_scores(self.scores, self.display_name, self.location)
 
+    def save(self, student, subsection, course):
+        """
+        Persist the SubsectionGrade.
+        """
+        visible_blocks = [
+            BlockRecord(unicode(location), weight, score.possible)
```

Comment:
Right - I had thought that a BlockRecord was the far edge, containing already-serialized data. See this test as an example: https://github.com/edx/edx-platform/blob/master/lms/djangoapps/grades/tests/test_models.py#L64-L80.

I'll se what I can do about making that assumption more explicit.